### ISSUE TITLE
Improve Loss Function and Implement Linear Regression

### ISSUE THREAD

**Issue by chrisssz** [jacksonmowry/puddle#2]
Replace current loss function with MSE

Implement linear regression and gradient descent to allow for backprop
